From: Jan Kara <jack@suse.cz>
Subject: printk: Make printing kthread use default priority
Patch-mainline: No, still being discussed upstream
References: bnc#744692, bnc#789311

Upstream has concerns that printing with RT priority can hog CPU and cause
issues and therefore they decided to switch the scheduling policy back to
normal.

Signed-off-by: Jan Kara <jack@suse.cz>

---
 kernel/printk/printk.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/kernel/printk/printk.c
+++ b/kernel/printk/printk.c
@@ -2769,9 +2769,6 @@ static int printk_kthread_func(void *dat
 static int __init_printk_kthread(void)
 {
 	struct task_struct *thread;
-	struct sched_param param = {
-		.sched_priority = MAX_RT_PRIO - 1,
-	};
 
 	if (!printk_kthread_can_run || printk_sync || printk_kthread)
 		return 0;
@@ -2783,7 +2780,6 @@ static int __init_printk_kthread(void)
 		return PTR_ERR(thread);
 	}
 
-	sched_setscheduler(thread, SCHED_FIFO, &param);
 	printk_kthread = thread;
 	return 0;
 }
